spring-pulsar/src/test/java/org/springframework/pulsar/core/reactive/DefaultReactiveMessageSenderFactoryTests.java

```diff
@@ -0,0 +1,159 @@
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
+		assertSenderHasTopicAndRouter("topic1", router);
+	}
+
+	@Test
+	void createProducerWithDefaultTopic() {
+		MutableReactiveMessageSenderSpec senderSpec = new MutableReactiveMessageSenderSpec();
+		senderSpec.setTopicName("topic0");
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				senderSpec);
+		ReactiveMessageSender<String> sender = senderFactory.createReactiveMessageSender(null, schema);
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
+		assertSenderHasTopicAndRouter("topic0", null);
+	}
+
+	@Test
+	void createProducerWithDefaultTopicAndMessageRouter() {
+		MutableReactiveMessageSenderSpec senderSpec = new MutableReactiveMessageSenderSpec();
+		senderSpec.setTopicName("topic0");
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				senderSpec);
+		MessageRouter router = mock(MessageRouter.class);
+		ReactiveMessageSender<String> sender = senderFactory.createReactiveMessageSender(null, schema, router);
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
+		assertSenderHasTopicAndRouter("topic0", router);
+	}
+
+	@Test
+	void createProducerWithSingleProducerCustomizer() {
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				null);
+		ReactiveMessageSenderBuilderCustomizer<String> customizer = builder -> builder.topic("topic1");
+		ReactiveMessageSender<String> sender = senderFactory.createReactiveMessageSender("topic0", schema, null,
+				Collections.singletonList(customizer));
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
+		assertSenderHasTopicAndRouter("topic1", null);
+	}
+
+	@Test
+	void createProducerWithMultipleProducerCustomizer() {
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				null);
+		ReactiveMessageSenderBuilderCustomizer<String> customizer1 = builder -> builder.topic("topic1");
+		MessageRouter router = mock(MessageRouter.class);
+		ReactiveMessageSenderBuilderCustomizer<String> customizer2 = builder -> builder.messageRouter(router);
+		ReactiveMessageSender<String> sender = senderFactory.createReactiveMessageSender("topic0", schema, null,
+				Arrays.asList(customizer1, customizer2));
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
+		assertSenderHasTopicAndRouter("topic1", router);
+	}
+
+	@Test
+	void createProducerWithNoTopic() {
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				null);
+		assertThatThrownBy(() -> senderFactory.createReactiveMessageSender(null, schema))
```

Comment:
**[TinyNit / Tip]** I just learned this more specific AssertJ API (you can tell I just learned it because I have not used it anywhere in this project yet) but... check out 
```java
assertThatIllegalArgumentException()
		.isThrownBy(() -> senderFactory.createReactiveMessageSender(null, schema))
		.withMessageContaining("Topic must be specified when no default topic is configured");
```

Reply:
👍

Reply:
done